packages/app-desktop/gui/MainScreen.tsx

```diff
@@ -312,12 +313,26 @@ class MainScreenComponent extends React.Component<Props, State> {
 	}
 
 	public updateRootLayoutSize() {
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
-		this.updateMainLayout(produce(this.props.mainLayout, (draft: any) => {
-			const s = this.rootLayoutSize();
-			draft.width = s.width;
-			draft.height = s.height;
-		}));
+		const layout = this.props.mainLayout;
+		if (!layout) return;
+
+		const newSize = this.rootLayoutSize();
+		const currentWidth = layout.width;
+		const currentHeight = layout.height;
+
+		if (!currentWidth || !currentHeight) {
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+			this.updateMainLayout(produce(layout, (draft: any) => {
+				draft.width = newSize.width;
+				draft.height = newSize.height;
+			}));
+			return;
+		}
+
+		const widthRatio = newSize.width / currentWidth;
+		const heightRatio = newSize.height / currentHeight;
+
+		this.updateMainLayout(scaleLayoutItemSizes(layout, newSize, widthRatio, heightRatio));
```

Comment:
```suggestion
		if (!currentWidth || !currentHeight) {
			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
			this.updateMainLayout(produce(layout, (draft: any) => {
				draft.width = newSize.width;
				draft.height = newSize.height;
			}));
		} else {
			const widthRatio = newSize.width / currentWidth;
			const heightRatio = newSize.height / currentHeight;

			this.updateMainLayout(scaleLayoutItemSizes(layout, newSize, widthRatio, heightRatio));
		}
```

**Optional/style**: Consider rewriting this as an `if`/`else` (removing the early return).